### PR TITLE
Pokenav Data

### PR DESCRIPTION
**Pull request by garakmon**

### CODE REVIEW

File: src/data/text/trainers_eye_messages.h [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+const u8 gUnknown_08621F65[] = _("Harness strong abilities.");
+const u8 gUnknown_08621F7F[] = _("The raw power of DRAGONS!");
+const u8 gUnknown_08621F99[] = _("I dedicate myself to the");
+const u8 gUnknown_08621FB2[] = _("POKéMON that saved me.");
+const u8 gUnknown_08621FC9[] = _("Dignity and respect.");
+const u8 gUnknown_08621FDE[] = _("I prefer POKéMON of grace.");
+const u8 gUnknown_08621FF9[] = _("I represent beauty as");
+const u8 gUnknown_0862200F[] = _("well as intelligence.");
+
+const u8 *const gUnknown_08622028[][4] =
```
Comment (PikalaxALT) [inline src/data/text/trainers_eye_messages.h]: Define a macro for each of these:
```c
#define MCFLAVOR(name) {gMatchCallFlavorText_##name##_Strategy, ... }
```

Comment (PikalaxALT) [reply src/data/text/trainers_eye_messages.h]: Index these with the trainer rematch enums:
```c
    [REMATCH_ROSE] = MCFLAVOR(AromaLadyRose),
```

File: src/pokenav_unk_3.c [hunk 2]
```diff
@@ -37,9 +37,10 @@ static u32 sub_81CACF8(struct Pokenav3Struct *);
 static u32 sub_81CAD20(s32);
 static bool32 sub_81CB1D0(void);
 
-extern const u8 gUnknown_08622508[];
-extern const u8 gUnknown_0862250A[];
-extern const u8 *const gUnknown_08622028[][4];
+#include "data/text/trainers_eye_messages.h"
+
+const u8 gUnknown_08622508[] = {0, 2};
+const u8 gUnknown_0862250A[] = {0, 1, 2, 0, 0, 0};
```
Comment (PikalaxALT) [inline src/pokenav_unk_3.c]: These trailing 0s may be little more than padding (0862250D-->08622510). See if it compiles if you remove them.

File: src/pokenav_unk_2.c [hunk 3]
```diff
@@ -1,6 +1,261 @@
...
+        .charBaseIndex = 3,
+        .mapBaseIndex = 0x1F,
+        .screenSize = 0,
+        .paletteMode = 0,
+        .priority = 3,
+        .baseTile = 0
+    }
+};
+
+u32 sub_81C9C6C(int);
```
Comment (PikalaxALT) [inline src/pokenav_unk_2.c]: For consistency's sake, LoopedTask should be `u32 func(s32)`. Some newer versions of GCC complain if you declare s32 as int instead.

File: src/pokenav_unk_9.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    .bg = 1,
+    .tilemapLeft = 1,
+    .tilemapTop = 6,
+    .width = 7,
+    .height = 2,
+    .paletteNum = 1,
+    .baseBlock = 20
+};
+
+// TODO: these are probably not strings?
```
Comment (PikalaxALT) [inline src/pokenav_unk_9.c]: They definitely look like strings. Color specs for gender symbols.

**Comment by garakmon** [comment]
updated.
fixed everything but the gitattributes.
I would have thought the line `*.bin = binary` would prevent the diffs from appearing like that, so idk what the issue is or how to resolve it.

File: src/data/text/match_call_messages.h [hunk 5]
```diff
@@ -0,0 +1,476 @@
+#define MCFLAVOR(name) {gMatchCallFlavorText_##name##_Strategy, \
```
Comment (PikalaxALT) [inline src/data/text/match_call_messages.h]: I can't read this macro name and not laugh

Comment (DizzyEggg) [reply src/data/text/match_call_messages.h]: lmao

**Pull request merged**